Standard.AI.Data.EntityIntelligence/Brokers/Data/DataBroker.cs

```diff
@@ -0,0 +1,56 @@
+﻿// ---------------------------------------------------------------------------------- 
+// Copyright (c) The Standard Organization, a coalition of the Good-Hearted Engineers 
+// ----------------------------------------------------------------------------------
+
+using Standard.AI.Data.EntityIntelligence.Models.Data;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+using System.Data.SqlClient;
+using Dapper;
+
+namespace Standard.AI.Data.EntityIntelligence.Brokers.DataStorages
+{
+    public class DataBroker : IDataBroker
+    {
+        private readonly DataStorageConfiguration configuration;
+
+        public DataBroker(DataStorageConfiguration configuration)
+        {
+            this.configuration = configuration;
+        }
+
+        public async Task<IEnumerable<TableColumnMetadata>> SelectSqlTablesMetadataAsync()
+        {
+            const string query = 
+                $@"SELECT 
```

Comment:
Milan, I'm new to the Standard but I believe having the actual query here is a business logic that shouldn't live in the broker. The broker must only pass the query to SQL Server and get back results.

Reply:
Hassan can weigh in here, I may be mistaken. My thinking is in the future we may need to pass a query to select 5/10/15 tables only not retrieve all tables to overcome the token limit. This means the SelectSqlTablesMetadataAsync should get the query as a parameter and the query be passed from a higher order service such as a Processing service.

Reply:
Good point, I also stopped there for a moment to think whether I am going into the right direction.

IMO Broker does what it's meant to. It hides external dependency from us, this is just another SELECT that we would have with LINQ just in a bit more detail. 
Only parts that I see as problematic are `CASE WHEN .... ` where there is some actual calculation. But I think that it should not be leaving the boundaries of a Broker. But even if we test that we're testing an external dependency. 

What I would usually do if this was a repository, I would make an integration test that would test the integration with the database and this mapping. I know that Broker is much more generic and it's not doing just repository is supposed to. But again, worth to think around that.

Reply:
I agree - I think this query needs to be passed in where this broker will just execute it. So we can test drive the contents of that query

Reply:
Then `SelectSqlTablesMetadataAsync` doesn't make sense. We would have only ExecuteQuery on the broker level that accepts query?

Reply:
I actually agree with you, I can't find a reason why should we have the metadata as a separate broker method, ,at the end of the day, it is executing a select statement as well. Let's see what Hassan has in mind :)

Reply:
one more thing, I don't think the broker should deal with local models (the return type in this case), "dynamic" might be a better fit. It is the foundation service on top of this broker that should do the mapping.

Reply:
Yeah Foundation Service is the one doing the mapping between external world/models and local ones. 

So having in mind that the other Broker method is also going to execute the query we can't just return this model here discussed and reuse it.
I see it in couple of directions:
- ` IEnumerable<dynamic> ExecuteQuery(string query)` that will be used both when getting tables metadata and executing a query from an AI
- ` IEnumerable<T> ExecuteQuery(string query)` where consumer (FoundationService) tells what it expects and that is a external model ofc, then it mapps it to a local one.

Or keeping the original idea but with the query and violating the responsibilities maybe
`IEnumerable<TableColumnMetadata> SelectSqlTablesMetadataAsync(string query)`